test/Constraints/keypath.swift

```diff
@@ -31,6 +31,9 @@ class Demo {
 }
 
 // FIXME: This error is better than it was, but the diagnosis should break it down more specifically to 'here's type.
+// TODO(diagnostics): Fix this regression, we want the following error message ideally
```

Comment:
Not sure what the best way to address this regression is. Figured I'd post a version with the other review comments addressed and leave this in for the moment. @xedin Thoughts?

Reply:
It's not _incorrect_ to produce that diagnostic really :) Let me pull your changes locally and see what could be done, I think it should be possible to default type variables found in the left-hand side to `Any` which is going to take affect only if the only way they get bindings is from context.